zoo_frontend/src/pages/login/login.jsx

```diff
@@ -60,8 +64,25 @@ class Login extends Component {
 
     try {
       await this.props.api.login(email, password);
-      Router.push('/');
+      if (process.browser) {
+        Router.events.on('routeChangeError', (err) => {
+          console.error(err);
+          if (this.notificationsRef && this.notificationsRef.current) {
+            this.notificationsRef.showNotification('error', 'Application redirection error, please close browser and re-open');
+          }
+        }); // leaving this in, let's us see if there are weird /login redirect issues
+
+        Router.push('/'); // previous block checks if this was successfull or not
+      }
+      return;
     } catch (err) {
+      console.log(err.message);
+      if (err.message === 'Role not found') {
+        setTimeout(() => {
```

Comment:
is this settimeout necessary?

Reply:
sadly yeah, something is going on with the timing of this that I couldn't get the notification to fire properly unless it wasn't in this sequence. Maybe cause it's in a catch block? idk